### ISSUE TITLE
Not working as expected

### ISSUE THREAD

**Issue by Rudis1261** [marcushat/RollingCurlX#13]
I haven't been able to find the root cause yet, but there's something wrong with the class. When using 10 concurrent connections I get 79 responses back out of 88 requests. 

If I use 20 concurrent, then I get 69 response back out of 88 requests.
If I use 8 concurrent connections, then I get 81 responses back out of the 88 requested. 

Smelling funky.

**Comment by marcushat** [comment]
Can you please post minimal example code that isolates the problem? I use this class literally all the time and have never had this issue. Also do you control the server you're making the resquests to and what php version.

**Comment by Rudis1261** [comment]
Hey Marcus, 

I am not sure it was due to the alterations I made, or the code itself. Unfortunately, I got frustrated and deleted the code, and create a generic MultiCurler with the code from php.net.

I can explain what I was doing if it helps, but I won't be able to re-create it. I didn't need the callback method, but rather populated the response in a public variable defined in the class. Then I had a count for the requests and a count for the responses. And these were irregular, depending on the size I chose as my concurrency. For my use case I need to be able to send a payload in array form, which is then returned on response. This is because I have various different operations to perform on a data set. 

Sorry I couldn't be of more help.
